src/MongoDB.Driver.Core/Core/WireProtocol/CommandUsingCommandMessageWireProtocol.cs

```diff
@@ -313,6 +313,18 @@ private TCommandResult ProcessResponse(ConnectionId connectionId, CommandMessage
                     {
                         var errmsg = errmsgBsonValue.ToString();
                         message = string.Format("Command {0} failed: {1}.", commandName, errmsg);
+                        // https://jira.mongodb.org/browse/CSHARP-2678
+                        if (materializedDocument.TryGetValue("code", out var errorCode)
+                            && errorCode.ToInt32() == 20
+                            && errmsg.StartsWith("Transaction numbers"))
```

Comment:
Don't know how strongly you feel about consistent style but our style is to have the operator (&& in this case) on the previous line)

Reply:
Fixed